onnxruntime/hosting/executor.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef ONNXRUNTIME_HOSTING_EXECUTOR_H
+#define ONNXRUNTIME_HOSTING_EXECUTOR_H
+
+#include <google/protobuf/stubs/status.h>
+
+#include "core/framework/data_types.h"
+
+#include "environment.h"
+#include "predict.pb.h"
+
+namespace onnxruntime {
+namespace hosting {
+
+class Executor {
+ public:
+  explicit Executor(std::shared_ptr<HostingEnvironment> hosting_env) : env_(hosting_env) {}
+
+  // Prediction method
+  google::protobuf::util::Status predict(const std::string& name, const std::string& version, const std::string& request_id,
```

Comment:
1. Capitalize predict
2. Consider returning PredictResponse and dealing with errors inside that class

Reply:
I'd like to return the status to caller and put the error handling logic on the caller side, because the error handling may different between HTTP and gRPC.